vassal-app/src/main/java/VASSAL/build/module/Map.java

```diff
@@ -1640,30 +2066,52 @@ public void drawPieces(Graphics g, int xOffset, int yOffset) {
     g2d.setComposite(oldComposite);
   }
 
+  /**
+   * Draws all of our "Drawable" components. Standard examples include {@link CounterDetailViewer}s (aka Mouse-over Stack Viewers),
+   * {@link GlobalMap}s (aka Overview Maps), {@link LOS_Thread}s, {@link MapShader}s, and the {@link KeyBufferer} (to show which
+   * pieces are selected).
+   * @param g target graphics object
+   * @param aboveCounters true means we should draw only the drawables that go above the counters; false means we should draw only the ones that go below
+   */
   public void drawDrawable(Graphics g, boolean aboveCounters) {
     for (Drawable drawable : drawComponents) {
-      if (!(aboveCounters ^ drawable.drawAboveCounters())) {
```

Comment:
Wow, I have no idea why we wrote that in such an unobvious way.

Reply:
Probably somebody came in from some mental side-tangent of "did any bits change", and tricked themselves into thinking they needed to xor to find which ones changed